### PR TITLE
Code Insights: Add dashboard count visibility info for the code insights edit UI

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes: https://github.com/sourcegraph/sourcegraph/issues/29943
Designs: [Figma](https://www.figma.com/file/db3qb0UjYLo9RYFxQYjWXz/Show-how-many-dashboards-have-the-current-editing-insight-%2329943-Ready-for-dev?node-id=275%3A1628)

## Context

This PR adds a dashboard warning for the edit UI in case the current editable insight has been included in more than just one dashboard and editing of this insight will propagate to other dashboards.

## Wildcard

This PR's design includes a Note-like component. Similar to wildcard '<Alert />' component but not exactly the same (see [designs](https://www.figma.com/file/db3qb0UjYLo9RYFxQYjWXz/Show-how-many-dashboards-have-the-current-editing-insight-%2329943-Ready-for-dev?node-id=275%3A1628), there is no icon for example). But still, it makes sense to have this Note component in the wildcard cause @AlicjaSuska also told that this Note component would be useful for other places in the app. And actually, we already have places like this with Note-like components.

So I added another variant `note` for the `<Alert />` component @sourcegraph/frontend-platform see this commit https://github.com/sourcegraph/sourcegraph/pull/30892/commits/8dcde357da70ed970d6e1b458b4f71268ab79a67

Screenshot 2022-02-09 at 18 08 53

## Todo 
- [x] Extend Insight types with `dashboardRefCount` gql field
- [x] Update Edit UI pages 
- [x] Add a new `note` variant to Alert component in the wildcard

### CODE REVIEW

File: client/web/src/enterprise/insights/core/backend/gql-api/deserialization/create-insight-view.ts [hunk 1]
```diff
@@ -90,11 +92,12 @@ export const createInsightView = (insight: InsightViewNode): Insight => {
                 // In gql api we don't have this concept as visibility on FE.
                 // Insights have special system about visibility on BE only.
                 visibility: '',
+                dashboardReferenceCount: insight.dashboardReferenceCount,
             }
         }
 
         case 'PieChartInsightViewPresentation': {
-            // At the moment we BE doesn't have special fragment type for Lang Stats repositories.
+            // At the moment BE doesn't have a special fragment type for Lang Stats repositories.
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/core/backend/gql-api/deserialization/create-insight-view.ts:100]: A+ comment fix 🎉

File: client/web/src/enterprise/insights/core/types/insight/common.ts [hunk 2]
```diff
@@ -30,13 +30,14 @@ export enum InsightType {
 }
 
 /**
- * These fields are needed only for the code insight FE runtime logic and they are not stored
+ * These fields are needed only for the code insight FE runtime logic, and they are not stored
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/core/types/insight/common.ts:33]: Crushing it with these grammatical fixes 👍

File: client/web/src/enterprise/insights/pages/insights/creation/capture-group/types.ts [hunk 3]
```diff
@@ -27,8 +27,13 @@ export interface CaptureGroupFormFields {
     stepValue: string
 
     /**
-     * This settings stands for turn on/off all repos mode that means this insight
+     * This setting stands for turn on/off all repos mode that means this insight
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/insights/creation/capture-group/types.ts:30]: Got a real Max Perkins vibe going on bravo 🙌

**Review by unclejustin — APPROVED** [review]
Excellent clean-up with this one, bravo sir!

File: client/web/src/enterprise/insights/pages/CodeInsightsRootPage.module.scss [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.tabs {
+    width: 100%;
+    border-bottom: 1px solid var(--border-color-2);
```
[the remark this answers is not in the document]
Comment (oleggromov) [reply client/web/src/enterprise/insights/pages/CodeInsightsRootPage.module.scss:3]: Screenshot 2022-02-10 at 14 44 26

**Pull request merged**